### PR TITLE
add BoundServiceAccountTokenVolume upgrade test config

### PR DESCRIPTION
**Pull request by zshihang**
add a periodic job config for alpha feature BoundServiceAccountTokenVolume upgrade test.

the test is added in https://github.com/kubernetes/kubernetes/pull/94835.

### CODE REVIEW

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (liggitt) [inline config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:16]: is this testing an upgrade or just a regular e2e test?

**Comment by liggitt** [comment]
I'm a little confused about whether this is actually setting up an upgrade test... I expected something more like this:

https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/gcp/upgrade-gce.yaml#L18-L25

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml]: it is a upgrade test even though there is no upgrade_args specified.

since `--test_args="--ginkgo.focus=\[Feature:BoundServiceAccountTokenVolume\]"` will select the upgrade test for this feature, it will run the upgrade test framework. the first target would be `--extract=ci/latest-1.19` and the unspecified target would be `ci/latest`.

Comment (liggitt) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml]: I see... can we make the upgrade target more explicit in the job config?

Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml]: done

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (liggitt) [inline config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:9]: I'm ok with this to start until we get the job green, then we should probably lower it to ~1 time a day

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (liggitt) [inline config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:2]: not sure if there's a convention for naming these. I'd tend to prefix with sig-auth. Thoughts, @BenTheElder ?

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (liggitt) [inline config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:12]: double check this is the current kubekins image

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (liggitt) [inline config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:22]: what will this do once ci/latest is 1.21? Should we be using stable1 markers? cc @justaugustus

**Review by liggitt — COMMENTED** [review]
A few questions on the job name and whether we should be using pinned versions (latest-1.19) or floating versions (stable1 and latest), but looks good in general

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:9]: ack

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml]: prepended with sig-auth

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml]: done

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:23]: use ci/latest-1.19 and ci/latest so that the hash and coredns fix are included. we can change them later if appropriate.

Comment (liggitt) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:23]: Sounds good, let's make sure that's in an issue to track

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (liggitt) [inline config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:10]: Oh, I forgot, we want to set up notifications for this job. This can be individuals until we get the job green, then should transition to the sig-auth list. See use of testgrid-alert-email in other jobs for examples, and include the mailing list transition in the follow-up tracking issue

Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:14]: done

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:28]: done. https://github.com/kubernetes/test-infra/issues/19575

**Comment by liggitt** [comment]
/lgtm

To testing reviewers/approvers

**Review by spiffxp — APPROVED** [review]
/approve

**Comment by liggitt** [comment]
> /hold for #19353 to merge

@dims... that's a self-reference... is the thing you meant to refer to merged? :-)

**Comment by zshihang** [comment]
/hold cancel

the test is cherry picked into 1.19

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (spiffxp) [inline config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:22]: I missed this... these are bootstrap.py parameters, they shouldn't be here if `decorate: true` is being used.

It's more preferable to use `decorate: true` than to revert to bootstrap.py, as bootstrap has been deprecated for years

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (spiffxp) [inline config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:25]: This isn't consumed by kubetest, you would want to set this as an `env:` on the container instead

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (spiffxp) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:22]: See https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml#L73-L82 for an example of what `command:` and `args:` should be instead

**Review by spiffxp — COMMENTED** [review]
I missed that this was using `decorate: true` and bootstrap.py at the same time.  I left review comments on how to migrate away from bootstrap.  I haven't personally tested them, so if you'd rather drop `decorate: true` that's fine too.

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:25]: done in the followup

File: config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml
Comment (zshihang) [reply config/jobs/kubernetes/sig-auth/serviceaccount-admission-controller-migration-config.yaml:22]: done. thank you for the review.

**Comment by dims** [comment]
> > /hold for #19353 to merge
> 
> @dims... that's a self-reference... is the thing you meant to refer to merged? :-)

Whoops!

**Pull request merged**
